lionweb/src/main/kotlin/com/strumenta/kolasu/lionweb/LionWebModelConverter.kt

```diff
@@ -183,16 +163,26 @@ class LionWebModelConverter(
 
                         is Reference -> {
                             if (feature == StarLasuLWLanguage.ASTNodeOriginalNode) {
-                                if (kNode.origin is KNode) {
-                                    val targetID = myIDManager.nodeId(kNode.origin as KNode)
-                                    lwNode.setReferenceValues(
-                                        StarLasuLWLanguage.ASTNodeOriginalNode,
-                                        listOf(
-                                            ReferenceValue(ProxyNode(targetID), null)
+                                val origin = kNode.origin
+                                if (origin is KNode) {
+                                    val targetID = myIDManager.nodeId(origin)
+                                    setOriginalNode(lwNode, targetID)
+                                } else if (origin is MissingASTTransformation) {
+                                    if (lwNode is DynamicNode) {
```

Comment:
Perhaps here we can check if it extends `AbstractClassifierInstance`, as it is a weaker request